docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,286 @@
+        - [ ] Verify the transaction (and the block it is included in) is invalid by providing the exact intrinsic gas minus one as `gas_limit` value to the transaction with all multiple combinations of values to the field.
+- [ ] Encoding Tests
+    - [ ] Verify correct transaction rejection due to incorrect field sizes
+- [ ] RPC Tests
+    - [ ] * Verify `eth_estimateGas` behavior for different valid combinations of the new transaction type
+    - [ ] Verify `eth_sendRawTransaction` using `execute`
+- [ ] Out-of-bounds checks
+    - [ ] Verify if the transaction has out-of-bounds conditions in its fields and verify:
+        - [ ] Max value for each field
+        - [ ] Max value + 1 for each field
+- [ ] Contract creation
+    - [ ] Verify that the transaction can create new contracts, or that it fails to do so if it's not allowed
+- [ ] Sender account modifications
+    - [ ] Verify that the sender account of the new transaction type transaction has its nonce incremented by one after the transaction is included in a block
+    - [ ] Verify that the sender account of the new transaction type transaction has its balance reduced by the correct amount (gas consumed and value) after the transaction is included in a block
+- [ ] Fork transition
+    - [ ] Verify that a block prior to fork activation where the new transaction type is introduced and containing the new transaction type is invalid.
+
+
+* Tests must be added to [`execution-apis`](https://github.com/ethereum/execution-apis) repository.
+
+### Framework Changes
+
+- [ ] Modify `transaction_intrinsic_cost_calculator` in the fork where the EIP is introduced in `src/ethereum_test_forks/forks/forks.py`, adding the appropriate new fields that the transaction introduced and the logic to the intrinsic gas cost calculation, if any.
+- [ ] Add the transaction type number to `tx_types` response in the fork where the EIP is introduced in `src/ethereum_test_forks/forks/forks.py` (If applicable add also to `contract_creating_tx_types`).
+
+
+## New Block Header Field
+
+### Test Vectors
+
+- [ ] Genesis value
+    - [ ] Verify, if possible, that the value can be set at genesis if the network starting fork is the activation fork, and that clients can consume such genesis.
+- [ ] Value behavior
+    - [ ] Verify, given multiple initial values, that the value is correctly modified for the current and subsequent blocks as expected, depending on the circumstances that affect the value as defined in the EIP.
+- [ ] Fork transition
+    - [ ] Verify initial value of the field at the first block of the activation fork.
+    - [ ] Verify that a block containing the new header field before the activation of the fork is invalid.
+    - [ ] Verify that a block lacking the new header field at the activation of the fork is invalid.
```

Comment:
if field is a list, verify the validity status of empty list, and max_allowed_size +/-1